src/Tools/MetricsOutputWriter.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeMetrics;
+
+namespace Metrics
+{
+    internal static class MetricsOutputWriter
+    {
+        private const string Version = "1.0";
+
+        public static void WriteMetricFile(ImmutableArray<(string, CodeAnalysisMetricData)> data, XmlTextWriter writer)
+        {
+            writer.Formatting = Formatting.Indented;
+
+            writer.WriteStartDocument();
+            writer.WriteStartElement("CodeMetricsReport");
+            writer.WriteAttributeString("Version", Version);
+            writer.WriteStartElement("Targets");
+            writeMetrics();
+            writer.WriteEndElement();
+            writer.WriteEndElement();
+            writer.WriteEndDocument();
+
+            return;
+
+            void writeMetrics()
+            {
+                foreach (var kvp in data)
+                {
+                    string filePath = kvp.Item1;
+                    CodeAnalysisMetricData metric = kvp.Item2;
+
+                    writer.WriteStartElement("Target");
+                    writer.WriteAttributeString("Name", Path.GetFileName(filePath));
+
+                    WriteMetricData(metric, writer);
+
+                    writer.WriteEndElement();
+                }
+            }
+        }
+
+        private static void WriteMetricData(CodeAnalysisMetricData data, XmlTextWriter writer)
+        {
+            writeHeader();
+            writeMetrics();
+            writeChildren();
+            writer.WriteEndElement();
+
+            return;
+
+            void writeHeader()
+            {
+                writer.WriteStartElement(data.Symbol.Kind.ToString());
+                switch (data.Symbol.Kind)
+                {
+                    case SymbolKind.NamedType:
+                        writer.WriteAttributeString("Name", data.Symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat));
```

Comment:
isn't MinimallyQualifiedFormat supposed to be only used with ToMinimalDisplayString?

Reply:
That API needs semantic model, and the current approach seem to work.

Reply:
asked because some of the options in display format won't be used in ToDisplayString since MinimallyQualifedFormat is supposed to have options what to remove when something such as namespace qualifier is not needed at the current position since there is using at the top and etc. but anyway, won't hurt even if some parts are ignored.